planner/core/plan.go

```diff
@@ -408,6 +411,10 @@ type basePhysicalPlan struct {
 	self             PhysicalPlan
 	children         []PhysicalPlan
 	cost             float64
+
+	// used by the new cost interface
+	planCostInit bool
```

Comment:
We only calculate the plan cost once and then always reuse it. So it contains the assumption that we don't modify the plan(or its children) once the plan cost is calculated for the first time. Currently we don't break the assumption but I think we should emphasize it somewhere because it is kind of error prone.